### PR TITLE
Use attachment to tag backquoted trees

### PR DESCRIPTION
**Pull request by odersky**

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/ast/untpd.scala [hunk 1]
```diff
@@ -337,7 +337,6 @@ object untpd extends Trees.Instance[Untyped] with UntypedTreeInfo {
   def Alternative(trees: List[Tree])(implicit src: SourceFile): Alternative = new Alternative(trees)
   def UnApply(fun: Tree, implicits: List[Tree], patterns: List[Tree])(implicit src: SourceFile): UnApply = new UnApply(fun, implicits, patterns)
   def ValDef(name: TermName, tpt: Tree, rhs: LazyTree)(implicit src: SourceFile): ValDef = new ValDef(name, tpt, rhs)
-  def BackquotedValDef(name: TermName, tpt: Tree, rhs: LazyTree)(implicit src: SourceFile): ValDef = new BackquotedValDef(name, tpt, rhs)
   def DefDef(name: TermName, tparams: List[TypeDef], vparamss: List[List[ValDef]], tpt: Tree, rhs: LazyTree)(implicit src: SourceFile): DefDef = new DefDef(name, tparams, vparamss, tpt, rhs)
   def BackquotedDefDef(name: TermName, tparams: List[TypeDef], vparamss: List[List[ValDef]], tpt: Tree, rhs: LazyTree)(implicit src: SourceFile): DefDef = new BackquotedDefDef(name, tparams, vparamss, tpt, rhs)
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/ast/untpd.scala]: Looks like this one was accidentally kept.

**Comment by smarter** [comment]
What's the motivation for this change ?

**Comment by odersky** [comment]
The motivation is to simplify. We get more and more Backquoted subclasses (quoted patterns would add a `BackquotedTypeDef` class), which leads to needless duplication of code.

**Comment by odersky** [comment]
> LGTM, but have you considered using a NameKind to represent backquoting instead ?

I did. It messes up other things. In particular, Name equality (`x` and x are the same name).


Yes, we did consider it.

**Pull request merged**